ts/sagas/wallet.ts

```diff
@@ -308,28 +305,35 @@ function* startOrResumeAddCreditCardSaga(
     // Even though this step is optional, in practice Pagopa will always
     // require the 3DS checkout for cards that gets added to the wallet.
     //
-
     const urlCheckout3ds =
       state.creditCardVerification.value.data.urlCheckout3ds;
-    const pagoPaToken = pmSessionManager.get();
-
-    if (pot.isNone(state.creditCardCheckout3ds)) {
-      if (urlCheckout3ds !== undefined && pagoPaToken.isSome()) {
-        yield put(
-          creditCardCheckout3dsRequest({
-            urlCheckout3ds,
-            paymentManagerToken: pagoPaToken.value
-          })
-        );
-        yield take(getType(creditCardCheckout3dsSuccess));
-        // all is ok, continue to the next step
-        continue;
-      } else {
-        // if there is no need for a 3ds checkout, simulate a success checkout
-        // to proceed to the next step
-        yield put(creditCardCheckout3dsSuccess("done"));
-        continue;
+    try {
+      // Request a new token to the PM. This prevent expired token during the webview navigation.
+      // If the request for the new token fails a new Error is catched, the step fails and we exit the flow.
+      const pagoPaToken = yield call(pmSessionManager.getNewToken());
```

Comment:
```suggestion
      const pagoPaToken : Option<PaymentManagerToken> = yield call(pmSessionManager.getNewToken);
```

change 
from
```typescript
public getNewToken = () => async () => {
```
to
```typescript
public getNewToken = async () : Promise<Option<T>> => {
```

Reply:
Fixed in [4ea211a](https://github.com/pagopa/io-app/pull/2490/commits/4ea211ab9dc9c9e7e004b53c646fbd4cf10c6050)